components/user_limit_modal/user_limit_modal.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Modal, Button} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import './user_limit_modal.scss';
+
+import upgradeImage from './user_limit_upgrade.svg';
+
+type Props = {
+    show: boolean;
+    actions: {
+        closeModal: () => void;
+    };
+};
+
+export default function UserLimitModal(props: Props) {
+    const onSubmit = () => {
+        // This does nothing until implementation of the upgrade tier page is complete
+        // Eventually, we will dispatch props.actions.openModal here
+    };
+
+    const close = () => {
+        props.actions.closeModal();
+    };
+
+    return (
+        <>
+            {props.show &&
+            <Modal
+                className={'UserLimitModal'}
+                show={props.show}
+                id='userLimitModal'
+                role='dialog'
+                onHide={close}
+            >
+                <Modal.Header closeButton={true}/>
+                <Modal.Body>
+                    <img
+                        className='upgrade-image'
+                        src={upgradeImage}
+                    />
+                    <div className='title'>{"You've reached the user limit"}</div>
```

Comment:
Nit: single-quotes instead of double-quotes

Reply:
Realized these should be translated anyways. Fixed.